pkg/blockstorage/azure/azuredisk.go

```diff
@@ -186,9 +187,12 @@ func (s *adStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.
 
 	var copyOptions *storage.CopyOptions
 	if t, ok := ctx.Deadline(); ok {
-		time := uint(t.Second())
+		time := time.Until(t).Seconds()
```

Comment:
Oh no!  Do you realize you've clobbered the "time" import variable? BTW, it is perfectly legal Go but a real **Go**tcha...